x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/persistence/ElasticsearchMappings.java

```diff
@@ -95,6 +110,278 @@
+                    .startObject(PROPERTIES)
+                        .startObject(CategorizationAnalyzerConfig.CATEGORIZATION_ANALYZER.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        // TOKENIZER, TOKEN_FILTERS and CHAR_FILTERS are complex types, don't parse or index
+                        .startObject(CategorizationAnalyzerConfig.TOKENIZER.getPreferredName())
+                            .field(ENABLED, false)
+                        .endObject()
+                        .startObject(CategorizationAnalyzerConfig.TOKEN_FILTERS.getPreferredName())
+                            .field(ENABLED, false)
+                        .endObject()
+                        .startObject(CategorizationAnalyzerConfig.CHAR_FILTERS.getPreferredName())
+                            .field(ENABLED, false)
+                        .endObject()
+                    .endObject()
+                .endObject()
+                .startObject(AnalysisConfig.LATENCY.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+                .startObject(AnalysisConfig.SUMMARY_COUNT_FIELD_NAME.getPreferredName())
+                    .field(TYPE, KEYWORD)
+                .endObject()
+                .startObject(AnalysisConfig.DETECTORS.getPreferredName())
+                    .startObject(PROPERTIES)
+                        .startObject(Detector.DETECTOR_DESCRIPTION_FIELD.getPreferredName())
+                            .field(TYPE, TEXT)
+                        .endObject()
+                        .startObject(Detector.FUNCTION_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.FIELD_NAME_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.BY_FIELD_NAME_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.OVER_FIELD_NAME_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.PARTITION_FIELD_NAME_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.USE_NULL_FIELD.getPreferredName())
+                            .field(TYPE, BOOLEAN)
+                        .endObject()
+                        .startObject(Detector.EXCLUDE_FREQUENT_FIELD.getPreferredName())
+                            .field(TYPE, KEYWORD)
+                        .endObject()
+                        .startObject(Detector.CUSTOM_RULES_FIELD.getPreferredName())
+                            .field(TYPE, NESTED)
```

Comment:
Do we need the nested document functionality here?  If not then `OBJECT` would be lighter-weight.  (Maybe at this level `NESTED` _is_ useful - I'm not sure.)